ansible_base/activitystream/apps.py

```diff
@@ -0,0 +1,38 @@
+from functools import partial
+
+from django.apps import AppConfig
+from django.db.models.signals import m2m_changed, post_save, pre_delete, pre_save
+
+from ansible_base.activitystream.signals import activitystream_create, activitystream_delete, activitystream_m2m_changed, activitystream_update
+
+
+def connect_activitystream_signals(cls):
+    post_save.connect(activitystream_create, sender=cls, dispatch_uid=f'dab_activitystream_{cls.__name__}_create')
+    pre_save.connect(activitystream_update, sender=cls, dispatch_uid=f'dab_activitystream_{cls.__name__}_update')
+    pre_delete.connect(activitystream_delete, sender=cls, dispatch_uid=f'dab_activitystream_{cls.__name__}_delete')
+
+    # Connect to m2m_changed signal for all m2m fields
+    for field in cls._meta.many_to_many:
+        if field.name in cls.activity_stream_excluded_field_names:
+            continue
+
+        fn = partial(activitystream_m2m_changed, field_name=field.name)
+        m2m_changed.connect(fn, sender=getattr(cls, field.name).through, dispatch_uid=f'dab_activitystream_{cls.__name__}_{field.name}_m2m_changed')
+
+
+class ActivitystreamConfig(AppConfig):
+    default_auto_field = 'django.db.models.BigAutoField'
+    name = 'ansible_base.activitystream'
+    label = 'dab_activitystream'
+    verbose_name = 'Activity Stream'
+
+    def ready(self):
+        # Iterate models and look for ones that inherit from AuditableModel.
+        # If found, connect the signal to the model by calling its static connect_signals().
+        from django.apps import apps
+
+        from ansible_base.activitystream.models import AuditableModel
+
+        for model in apps.get_models():
+            if issubclass(model, AuditableModel):
+                connect_activitystream_signals(model)
```

Comment:
It's not that there is anything wrong with this, but I would like to raise the fact that DAB RBAC & resource registry do use a "registry" which is a competing approach to do the same thing. I also agree that the RBAC & resource registry should eventually (perhaps partially) merged into a single registry. That argues for doing this as a registry for no other reason that there could be more consistency.

Reply:
I intentionally didn't want a registry here, just because it's another layer and it doesn't really buy us anything. I mean, I suppose it gives you a way to programmatically "back out" of having a model in the activity stream, but I can't imagine a scenario where that would be wanted.

If we had a "registry" I'd still want it to be automated and work like it does now (just that inheriting from `AuditableModel` would auto-register the model with the registry instead of directly connect the signals). But I just don't see the advantage.

It's also something we could do later on if there becomes a more compelling argument for it. The calling code shouldn't have to change at all if we do it like I said above.